greedy_method/knapsack_problem.py

```diff
@@ -0,0 +1,104 @@
+# To get an insight into Greedy Algorithm
+# Knapsack is a common Greedy Algo problem
+
+
+"""
+You have bags of wheat with different weights and a different profits for each.
+eg.
+profit 5 8 7 1 12 3 4
+weight 2 7 1 6 4  2 5
+max_Weight 100
+Calculate the max profit the shopkeeper can make for the given max_Weight that could be carried.
+"""
+
+# Importing copy module for deepcopy operations
+import copy
+
+
+def calc_Profit(profit: list, weight: list, max_Weight: int) -> int:
+    """
+    Function description is as follows-
+    :param profit: Take a list of profits
+    :param weight: Take a list of weight if bags corresponding to the profits
+    :param max_Weight: Max. weight that could be carried
+    :return: Max expected gain
+
+    >>> calc_Profit([1,2,3], [3,4,5], 15)
+    6
+    >>> calc_Profit([10, 9 , 8], [3 ,4 , 5], 25)
+    27
+
+    """
+
+    # List created to store profit gained for the 1kg in case of each weight
+    # respectively
+    profit_By_Weight = list()
+
+    # Calculate and append profit/weight for each
+    for i in range(len(weight)):
+        profit_By_Weight.append(profit[i] / weight[i])
+
+    # Creating a copy of the list and sorting proit/weight in ascending order
+    temp_PBW = sorted(copy.deepcopy(profit_By_Weight))
+
+    # declaring useful variables
+    length = len(temp_PBW)
+    limit = 0
+    gain = 0
+    i = 0
+
+    # loop till the total weight do not reach max limit i.e. 15 kg and till i<l
+    while limit <= 15 and i < length:
```

Comment:
```suggestion
    while i < limit <= 15:
```

Reply:
It may be ```i <= limit <= 15 ``` since initially both i and limit is 0.